runelite-client/src/main/java/net/runelite/client/plugins/StatusBars/StatusBarsOverlay.java

```diff
@@ -0,0 +1,307 @@
+import net.runelite.api.Point;
+import net.runelite.api.Skill;
+import net.runelite.api.VarPlayer;
+import net.runelite.api.Varbits;
+import net.runelite.api.queries.InventoryWidgetItemQuery;
+import net.runelite.api.widgets.Widget;
+import net.runelite.api.widgets.WidgetInfo;
+import net.runelite.api.widgets.WidgetItem;
+import net.runelite.client.plugins.itemstats.Effect;
+import net.runelite.client.plugins.itemstats.ItemStatChanges;
+import net.runelite.client.plugins.itemstats.StatChange;
+import net.runelite.client.plugins.itemstats.StatsChanges;
+import net.runelite.client.ui.FontManager;
+import net.runelite.client.ui.overlay.Overlay;
+import net.runelite.client.ui.overlay.OverlayLayer;
+import javax.inject.Inject;
+import net.runelite.client.game.SkillIconManager;
+import net.runelite.client.ui.overlay.OverlayPosition;
+import net.runelite.client.util.QueryRunner;
+import net.runelite.client.ui.overlay.components.TextComponent;
+
+public class StatusBarsOverlay extends Overlay
+{
+	// Static colors
+	private static final Color DEPLETED = new Color(25, 25, 25, 150);
+	private static final Color PRAYER_COLOR = new Color(50, 200, 200, 200);
+	private static final Color QUICK_PRAYER_COLOR = new Color(57, 255, 186, 225);
+	private static final Color BACKGROUND = new Color(0, 0 , 0, 150);
+	private static final Color HEALTH_COLOR = new Color(200, 35, 0, 150);
+	private static final Color POISONED_COLOR = new Color(0, 145, 0, 150);
+	private static final Color VENOMED_COLOR = new Color(0, 65, 0, 150);
+	private static final Color HEAL_COLOR = new Color (255, 112, 6, 150);
+	private static final Color PRAYER_HEAL_COLOR = new Color(57, 255, 186, 75);
+	private static final BufferedImage NO_IMAGE = new BufferedImage(17, 17, BufferedImage.TYPE_INT_ARGB);
+	private static final int HEIGHT = 252;
+	private static final int WIDTH = 20;
+	private static final int PADDING = 1;
+	private static final int IMAGE_SIZE = 16;
+	private static final int HEALTH_LOCATION_X = 2;
+	private static final int PRAYER_LOCATION_X = 3;
+	private final Client client;
+	private final StatusBarsConfig config;
+	private final SkillIconManager skillIconManager;
+	private final QueryRunner queryRunner;
+	private final ItemStatChanges itemStatChanges;
+	private final TextComponent textComponent = new TextComponent();
+
+	@Inject
+	private StatusBarsOverlay(Client client, StatusBarsConfig config, SkillIconManager skillIconManager, QueryRunner queryRunner, ItemStatChanges itemStatChanges)
+	{
+		setPosition(OverlayPosition.DYNAMIC);
+		setLayer(OverlayLayer.ABOVE_WIDGETS);
+		this.client = client;
+		this.config = config;
+		this.skillIconManager = skillIconManager;
+		this.queryRunner = queryRunner;
+		this.itemStatChanges = itemStatChanges;
+	}
+
+	@Override
+	public Dimension render(Graphics2D g)
+	{
+		// gets the current viewport and changes values accordingly so the bars are in the right position.
+		for (Viewport viewport : Viewport.values())
```

Comment:
Reduce the complexity of this method. Instead of doing everything in this giant outer loop, find correct widgets first, then process them. I would go for max 3 nested blocks here, otherwise its super hard to read and follow.